ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java

```diff
@@ -349,6 +354,90 @@ public boolean checkConnection(String url) {
     return (clientResp.getStatus() < HttpServletResponse.SC_UNAUTHORIZED);
   }
 
+  @Override
+  public List<RangerPolicy> addDenyPolicies(List<RangerPolicy> rangerPolicies, String rangerServiceName,
+                                            String sourceDb, String targetDb) {
+    List<RangerPolicy> rangerPoliciesToImport = new ArrayList<RangerPolicy>();
+    if (CollectionUtils.isNotEmpty(rangerPolicies)) {
+      for (RangerPolicy rangerPolicy : rangerPolicies) {
+        rangerPoliciesToImport.add(rangerPolicy);
+      }
+    }
+
+    RangerPolicy denyRangerPolicy = null;
+
+    if (sourceDb.endsWith("/")) {
+      sourceDb = StringUtils.removePattern(sourceDb, "/+$");
+    }
+    if (targetDb.endsWith("/")) {
+      targetDb = StringUtils.removePattern(targetDb, "/+$");
+    }
+    if (!StringUtils.isEmpty(rangerServiceName)) {
```

Comment:
Deny policy is added only when !StringUtils.isEmpty(rangerServiceName), this check should be done at the beginning?
Also, REPL_RANGER_SERVICE_NAME is a mandatory config during ranger repl dump. Shouldn't this be mandatory at load as well? Wondering how does service mapping work in that case.